### PR TITLE
Added in a limit to fix issue in motors and check for errors (Issue #390)

### PR DESCRIPTION
**Pull request by RohanCheri**
I have added in code to fix the issue with the motors and also account for errors (Issue #390)

### CODE REVIEW

File: OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java [hunk 1]
```diff
@@ -47,10 +50,33 @@ public String toString() {
          * @return reference returned so calls can be chained
          */
         public Builder<V> speed(Expr<V> speed) {
-            this.speed = speed;
+//            this.speed = speed;
```
Comment (boonto) [inline OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java]: You can remove this comment.

File: OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java [hunk 2]
```diff
@@ -47,10 +50,33 @@ public String toString() {
...
+                }
+                else if (val.compareTo(new BigDecimal(-100)) == -1){
+                    outVal = "100";
+                }
+                else{
+                    outVal = val.toString();
+                }
+
+                return NumConst.make(outVal, speed.getProperty(), speed.getComment());
+            }
```
Comment (boonto) [inline OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java]: catch has to be on the same line as the previous }

File: OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java [hunk 3]
```diff
@@ -47,10 +50,33 @@ public String toString() {
...
+            catch(Exception e){
```
Comment (boonto) [inline OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java]: Do not use a catch all exception, only catch the exception that may occur here.

File: OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/visitor/validate/AbstractProgramValidatorVisitor.java [hunk 4]
```diff
@@ -462,9 +463,17 @@ private void checkIfMotorRegulated(Phrase<Void> driveAction, ConfigurationCompon
...
-            if ( Integer.valueOf(speedNumConst.getValue()) == 0 ) {
-                action.addInfo(NepoInfo.warning("MOTOR_SPEED_0"));
-                this.warningCount++;
+            try {
+                if (new BigInteger(speedNumConst.getValue()).compareTo(new BigInteger("0")) == 0) {
+                    action.addInfo(NepoInfo.warning("MOTOR_SPEED_0"));
+                    this.warningCount++;
+                }
+            }
+            catch(Exception e){
```
Comment (boonto) [inline OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/visitor/validate/AbstractProgramValidatorVisitor.java]: Same as in the other file.

File: OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java (hunk 3 above)
Comment (RohanCheri) [reply OpenRobertaRobot/src/main/java/de/fhg/iais/roberta/syntax/MotionParam.java]: When I was testing it, I found that multiple types of expressions were being acautght. I originally only caught null pointer but It still produced errors.

**Pull request closed without merging**